lib/connections/service.go

```diff
@@ -733,6 +739,21 @@ func (s *service) CommitConfiguration(from, to config.Configuration) bool {
 	return true
 }
 
+func (s *service) checkAndSignalConnectLoopOnUpdatedDevices(from, to config.Configuration) {
+	oldDevices := make(map[protocol.DeviceID]config.DeviceConfiguration, len(to.Devices))
+	for _, dev := range from.Devices {
+		oldDevices[dev.DeviceID] = dev
+	}
+
+	for _, dev := range to.Devices {
+		oldDev, ok := oldDevices[dev.DeviceID]
+		if !ok || !reflect.DeepEqual(oldDev.Addresses, dev.Addresses) {
```

Comment:
Using reflect for comparing string slices is grim. I'd expect something to exist in the standard library.

Reply:
I don't think that exists and we already did implement this super elaborate function in `api_test.go` as `equalStrings`. Could move that to the `lib/utils` package for everyone to use.

Reply:
Awesome, I'll move it to lib/utils.